src/main/java/org/candlepin/subscriptions/marketplace/MarketplacePayloadMapper.java

```diff
@@ -0,0 +1,199 @@
+
+import org.candlepin.subscriptions.db.model.HardwareMeasurementType;
+import org.candlepin.subscriptions.db.model.ServiceLevel;
+import org.candlepin.subscriptions.db.model.Usage;
+import org.candlepin.subscriptions.files.ProductProfileRegistry;
+import org.candlepin.subscriptions.json.TallySnapshot;
+import org.candlepin.subscriptions.json.TallySummary;
+import org.candlepin.subscriptions.marketplace.api.model.UsageEvent;
+import org.candlepin.subscriptions.marketplace.api.model.UsageMeasurement;
+import org.candlepin.subscriptions.marketplace.api.model.UsageRequest;
+import org.candlepin.subscriptions.tally.UsageCalculation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Maps TallySummary to payload contents to be sent to RHM apis
+ */
+@Service
+public class MarketplacePayloadMapper {
+    private static final Logger log = LoggerFactory.getLogger(MarketplacePayloadMapper.class);
+
+    private final ProductProfileRegistry profileRegistry;
+
+    @Autowired
+    public MarketplacePayloadMapper(ProductProfileRegistry profileRegistry) {
+        this.profileRegistry = profileRegistry;
+    }
+
+    /**
+     * Create UsageRequest pojo to send to Marketplace
+     *
+     * @param tallySummary TallySummary
+     * @return UsageRequest
+     */
+    public UsageRequest createUsageRequest(TallySummary tallySummary) {
+
+        UsageRequest usageRequest = new UsageRequest();
+
+        var usageEvents = produceUsageEvents(tallySummary);
+        usageEvents.forEach(usageRequest::addDataItem);
+
+        log.debug("UsageRequest {}", usageRequest);
+
+        return usageRequest;
+    }
+
+    /**
+     * We only want to send snapshot information for OpenShift-metrics, OpenShift-dedicated-metrics product
+     * ids.  To prevent duplicate data, we don't want to send snapshots with the Usage or ServiceLevel of
+     * "_ANY".  We only want to report on hourly metrics, so the Granularity should be HOURLY.
+     *
+     * @param snapshot tally snapshot
+     * @return eligibility status
+     */
+    protected boolean isSnapshotPAYGEligible(TallySnapshot snapshot) {
+        String productId = snapshot.getProductId();
+        var applicableProducts = List.of("OpenShift-metrics", "OpenShift-dedicated-metrics");
+        boolean isApplicableProduct = applicableProducts.contains(productId);
+
+        boolean isHourlyGranularity = Objects.nonNull(snapshot.getGranularity()) &&
+            Objects.equals(TallySnapshot.Granularity.HOURLY, snapshot.getGranularity());
+
+        boolean isSpecificUsage = Objects.nonNull(snapshot.getUsage()) &&
```

Comment:
We can drop `nonNull` here too.  I would flip the constant to be the first argument just for aesthetic and consistency reasons though.

```java
public static boolean equals(Object a, Object b) {
    return (a == b) || (a != null && a.equals(b));
}
 ```

Reply:
There was some downstream issues with this at first because of the possibility of it being `__EMPTY` and still needing to be filtered out....but I did refactor to make it more aesthetically pleasing / readable